gpflow/models/sgpr.py

```diff
@@ -55,7 +55,7 @@ def upper_bound(self):
         x_data, y_data = self.data
         num_data = tf.cast(tf.shape(y_data)[0], default_float())
 
-        Kdiag = self.kernel(x_data)
+        Kdiag = self.kernel(x_data, full=False)
```

Comment:
Out of scope for this PR I guess but I'd much rather call the argument something else, `diag=True` perhaps (given that we want the default to be "full covariance" whereas in other parts of the code `full_cov=False` is the default).

Reply:
@awav , just flagging up.